src/function/scalar/string/string_split.cpp

```diff
@@ -0,0 +1,288 @@
+			// potential delimiter match
+			if (utf8proc_codepoint(&input[offset], cp_sz) == delim_cps[0] && offset + delim_size <= size) {
+				idx_t delim_offset = cp_sz;
+				for (idx_t i = 1; i < delim_cps.size(); i++) {
+					if (utf8proc_codepoint(&input[offset + delim_offset], cp_sz) != delim_cps[i])
+						break;
+					delim_offset += cp_sz;
+				}
+				// delimiter found: skip start over delimiter
+				if (delim_offset == delim_size) {
+					start = offset + delim_size;
+					return offset;
+				}
+			}
+		}
+		return offset;
+	}
+
+protected:
+	vector<utf8proc_int32_t> delim_cps;
+	const size_t delim_size;
+};
+
+struct RegexIterator : virtual public Iterator {
+public:
+	RegexIterator(size_t input_size, unique_ptr<RE2> re, const bool ascii_only)
+	    : Iterator(input_size), re(move(re)), ascii_only(ascii_only) {
+	}
+	idx_t Next(const char *input) override {
+		duckdb_re2::StringPiece input_sp(input, size);
+		duckdb_re2::StringPiece match;
+		if (re->Match(input_sp, start, size, RE2::UNANCHORED, &match, 1)) {
+			offset = match.data() - input;
+			// special case: 0 length match
+			if (match.size() == 0) {
+				if (ascii_only)
+					offset++;
+				else
+					offset = utf8proc_next_grapheme(input, size, offset);
+				start = offset;
+			} else {
+				start = offset + match.size();
+			}
+		} else {
+			offset = size;
+		}
+		return offset;
+	}
+
+protected:
+	unique_ptr<RE2> re;
+	const bool ascii_only;
+};
+
+void string_split(const char *input, Iterator &iter, ChunkCollection &result) {
+	auto append_chunk = make_unique<DataChunk>();
+	vector<LogicalType> types = {LogicalType::VARCHAR};
+	append_chunk->Initialize(types);
+
+	// special case: empty string
+	if (iter.size == 0) {
+		FlatVector::GetData<string_t>(append_chunk->data[0])[append_chunk->size()] =
+		    StringVector::AddString(append_chunk->data[0], &input[0], 0);
+		append_chunk->SetCardinality(append_chunk->size() + 1);
+	}
+
+	while (iter.HasNext()) {
+		if (append_chunk->size() == STANDARD_VECTOR_SIZE) {
+			result.count += append_chunk->size();
+			result.chunks.push_back(move(append_chunk));
+			append_chunk = make_unique<DataChunk>();
+			append_chunk->Initialize(types);
+		}
+
+		idx_t start = iter.Start();
+		idx_t end = iter.Next(input);
+		size_t length = end - start;
+
+		FlatVector::GetData<string_t>(append_chunk->data[0])[append_chunk->size()] =
+		    StringVector::AddString(append_chunk->data[0], &input[start], length);
+		append_chunk->SetCardinality(append_chunk->size() + 1);
+	}
+	if (append_chunk->size() > 0) {
+		result.count += append_chunk->size();
+		result.chunks.push_back(move(append_chunk));
+	}
+	result.Verify();
+}
+
+unique_ptr<ChunkCollection> string_split(string_t input, string_t delim, const bool regex) {
+	const char *input_data = input.GetData();
+	size_t input_size = input.GetSize();
+	const char *delim_data = delim.GetData();
+	size_t delim_size = delim.GetSize();
+
+	bool ascii_only = true;
+	for (auto i = 0; i < (int)input_size; ++i) {
```

Comment:
We have code for this check in `Utf8Proc::Analyze`

Reply:
Changed to use that instead